src/components/apps/launch/AppLaunchForm.js

```diff
@@ -427,6 +441,26 @@ const AppLaunchForm = (props) => {
             )
         );
 
+    const stepperRef = React.useRef(null);
+    const [stepperHeight, setStepperRef] = useComponentHeight();
+
+    const bottomNavRef = React.useRef(null);
+    const [bottomNavHeight, setBottomNavRef] = useComponentHeight();
+
+    const appBarHeight =
+        stepperHeight +
+        appInfoHeight +
+        bottomNavHeight +
+        (isMobile ? MOBILE_NAVIGATION_BAR_HEIGHT : NAVIGATION_BAR_HEIGHT);
+
+    React.useEffect(() => {
+        setStepperRef(stepperRef);
+    }, [stepperRef, setStepperRef]);
+
+    React.useEffect(() => {
+        setBottomNavRef(bottomNavRef);
+    }, [bottomNavRef, setBottomNavRef]);
+
```

Comment:
Calculate `Height` for `AppInfo, Stepper and Bottom Nav bar` dynamically.

Reply:
I don't believe these useEffects are needed.  `useComponentHeight` already has a useEffect for when the ref changes

Reply:
Feels like we talked about this before - https://github.com/cyverse-de/sonora/pull/126#discussion_r447908442

Reply:
Ah, I forgot about that convo. Does the same error happen here?  I'm unsure as to why that could be 😕

Reply:
Removing it actually pushed the nav bar out of bounds. Surprisingly I didn't get an exception in console 😕 . I am going to save the investigation step for a later time. For now am gonna leave the `useEffect` in place.

Reply:
Small update - putting the `setBottomNavRef(bottomNavRef);` outside the `useEffect` does result in `too many rerenders` error. When I wrote the previous commented out the whole `useEffect` without moving `setBottomNavRef(bottomNavRef);`.